src/cronMatcher.ts

```diff
@@ -1,16 +1,26 @@
 import {
   CRONTAB_NUMBER,
   CRONTAB_RANGE,
+  CRONTAB_TIME_PARTS,
   CRONTAB_WILDCARD,
 } from "./cronConstants";
-import { ParsedCronItem, TimestampDigest } from "./interfaces";
+import { TimestampDigest } from "./interfaces";
+
+// Crontab ranges from the minute, hour, day of month, month and day of week parts of the crontab line
+interface CrontabRanges {
+  minutes: number[];
+  hours: number[];
+  dates: number[];
+  months: number[];
+  dows: number[];
+}
```

Comment:
Please do the following:

1. Rename this from `CrontabRanges` to `ParsedCronMatch`
2. Move this to interfaces just above `ParsedCronItem`
3. Add `extends ParsedCronMatch` to the interface definition of `ParsedCronItem` 
4. Move the comments and definitions for these 5 items from `ParsedCronItem` to `ParsedCronMatch`

Reply:
good suggestion!